cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -231,3 +240,29 @@ func isServiceActive(name string) (bool, error) {
 	}
 	return initSystem.ServiceIsActive(name), nil
 }
+
+// MutateCgroupDriver can be called to set the KubeletConfiguration cgroup driver to systemd.
+// Currently this cannot be as part of Default() because the function is called for
+// upgrades too, which can break existing nodes after a kubelet restart.
+// TODO: https://github.com/kubernetes/kubeadm/issues/2376
+func MutateCgroupDriver(cfg *kubeadmapi.ClusterConfiguration) {
+	const driver = "systemd"
```

Comment:
Shall we consider using global constant instead of redefining the driver?
https://github.com/kubernetes/kubernetes/blob/be2a2988a0c6b5f0b66e4ebdc335fc1318e029ca/cmd/kubeadm/app/util/cgroupdriver.go#L29

Reply:
i prefer to contain it here.
actually there is already a constant in preflight for the driver in the
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks_linux.go#L35

but i'd like to remove that preflight and docker detection code, if we get this in 1.21:
https://github.com/kubernetes/kubernetes/issues/98222